tiatoolbox/models/engine/nucleus_instance_segmentor.py

```diff
@@ -0,0 +1,704 @@
+# ***** BEGIN GPL LICENSE BLOCK *****
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software Foundation,
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# The Original Code is Copyright (C) 2021, TIA Centre, University of Warwick
+# All rights reserved.
+# ***** END GPL LICENSE BLOCK *****
+
+"""This module enables nucleus instance segmentation."""
+
+import uuid
+from typing import Callable, List, Union
+
+# replace with the sql database once the PR in place
+import joblib
+import numpy as np
+import pygeos
+import torch
+import tqdm
+
+from tiatoolbox.models.engine.semantic_segmentor import (
+    IOSegmentorConfig,
+    SemanticSegmentor,
+    WSIStreamDataset,
+)
+from tiatoolbox.tools.patchextraction import PatchExtractor
+
+
+# till the day Python could natively pickle Object method/static method
```

Comment:
Can we make this comment shorter - it’s a bit long winded

Reply:
Not possible for me, you can paraphrase the following if you want. I think its the most compact I can write
```
# Python has yet being able to natively pickle Object method/static method.
# Only top-level function is passable to multi-processing as caller.
# May need 3rd party libraries to use method/static method otherwise.
```

Reply:
# Python is yet to be able to natively pickle Object method/static method.
# Only top-level function is passable to multi-processing as caller.
# May need 3rd party libraries to use method/static method otherwise.